core/state/statedb.go

```diff
@@ -798,6 +809,7 @@ func (s *StateDB) Prepare(thash, bhash common.Hash, ti int) {
 	s.thash = thash
 	s.bhash = bhash
 	s.txIndex = ti
+	s.accessList.Clear()
```

Comment:
Instead of clearing here, maybe it's more meaningful in `clearJournalAndRefund`? That is the one which recreates the journal too for the next tx. Would make more sense to keep similar things and mechanisms close by.

Reply:
I've removed `Clear()` now, and instead allocates a fresh one. This method is run _before_ executing the tx, as opposed to `clearJournal...`, so therefore I've let this stay here (but be an alloc instead)